From b3ac063e6020be6a103e9e9e7d468af9746b7eab Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Tue, 01 Feb 2022 18:54:32 +0200
Subject: [PATCH] [SW-69189] gaudi3: add placeholder for wait for signal

After the common initialization flow, die1 (slave die) is expected
to run while loop waiting for further signal from die0 (master die).
This patch add placeholder function for die1 waiting for signal.

Change-Id: I8ca33e1712d1291cc7e2bda10c3fd076633a97a7
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index 985b56b..5ba3fee 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -449,12 +449,14 @@
 }
 
 __weak void arcboot_rom_disable_spi_gw(void) {}
+__weak void arcboot_rom_wait_for_signal(void) {}
 
 int arcboot_rom_main(void)
 {
 	arcboot_rom_init();
 	arcboot_rom_fuse();
 	arcboot_rom_disable_spi_gw();
+	arcboot_rom_wait_for_signal();
 	arcboot_rom_in_recovery();
 	arcboot_rom_request_cpld_wdog();
 	arcboot_rom_bringup_pcie();
diff --git a/arch/gaudi3/arc/Makefile b/arch/gaudi3/arc/Makefile
index 2eb9f50..0f759fe 100644
--- a/arch/gaudi3/arc/Makefile
+++ b/arch/gaudi3/arc/Makefile
@@ -2,3 +2,4 @@
 
 BINARIES = arcboot_rom.bin
 ARCBOOT_COMMON_SRCS_C += $(shell ls $(HL_DEV_ARCH)/spi_flash.c)
+ARCBOOT_COMMON_SRCS_C += $(shell ls $(HL_DEV_ARCH)/slave.c)
diff --git a/arch/gaudi3/arc/slave.c b/arch/gaudi3/arc/slave.c
new file mode 100644
index 0000000..0d1e506
--- /dev/null
+++ b/arch/gaudi3/arc/slave.c
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier:     GPL-2.0+
+/*
+ * Copyright (C) 2022 HabanaLabs Ltd.
+ * All Rights Reserved.
+ */
+
+#include <arcboot.h>
+
+
+void arcboot_rom_wait_for_signal(void)
+{
+	if (!arcboot_rom_global.die_id)
+		return;
+
+	/*
+	 * TODO:
+	 * 1. poll on GPIO 76 for recovery signal set by master die.
+	 * 2. If set, call recovery flow, else sleep 1us and keep polling
+	 */
+}
